tlatools/org.lamport.tlatools/src/tlc2/util/SetOfStates.java

```diff
@@ -72,6 +80,22 @@ public final void clear() {
 		this.states = new TLCState[length];
 	}
 	
+	private final void grow(final ITool tool) {
+		final TLCState[] old = states;
+		this.count = 0;
+		this.length = 2 * this.length + 1;
+		this.thresh = this.length / 2;
+		this.states = new TLCState[this.length];
+		for (int i = 0; i < old.length; i++) {
+			final TLCState s = old[i];
+			// This is where we have to redundantly compute the state's
+			// fingerprint. Thus, try to minimize the number of grow operations.
+			if (s != null) {
+				this.put(s.fingerPrint(tool), s, tool);
+			}
+		}
+	}
+
```

Comment:
unless I'm missing something, tool parameter is unused. It is passed down to this.put, which in turn passes it down to grow(tool). This duplication seems unneeded

Reply:
Worker#doNextCheckLiveness > SetOfStates#put > SetOfStates#grow

Reply:
What SetOfStates#put does with its `tool` argument is to pass it down to SetOfStates#grow, and what grow does with it is to pass it to put. It's doesn't seem to be used otherwise

Reply:
Note `s.fingerPrint(tool)` on 94 above.

Reply:
ah thanks! Somehow I've missed that reference